### ISSUE TITLE
intermittent half close failures on trunk

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5558]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#5558|
|Type|defect|
|Created|2012-03-17 01:48:29Z|
|Branch|https://github.com/twisted/twisted/tree/half-close-failing-5558|

```
===============================================================================
[ERROR]
Traceback (most recent call last):
  File ".../twisted/test/test_tcp.py", line 1631, in testWriteCloseNotification
    f.protocol.transport.loseWriteConnection()
exceptions.AttributeError: 'NoneType' object has no attribute 'loseWriteConnection'

twisted.test.test_tcp.HalfCloseTestCase.testWriteCloseNotification
===============================================================================
[ERROR]
Traceback (most recent call last):
  File ".../twisted/test/test_tcp.py", line 1594, in _tearDown
    self.assertEqual(self.f.protocol.closed, 0)
exceptions.AttributeError: class RegisterTLSProtocol has no attribute 'closed'

twisted.test.test_tcp.HalfCloseTestCase.testWriteCloseNotification
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__5558 5558
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_half_close_failing_5558 branches/half-close-failing-5558
branch_author__itamar itamar
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1331948909000000 1331948909000000
changetime__1332246322000000 1332246322000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [33911]) Branching to 'half-close-failing-5558'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst removed owner|
|-|-|

Turns out `twisted.internet.test.test_newtls` had a minor bug.

buildbot.twistedmatrix.com/boxes-supported?branch=/branches/half-close-failing-5558

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @itamarst|
|-|-|

Oh my goodness that is terrible.  And the fix is obviously correct.  Please land.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [33958]) Merge half-close-failing-5558: Fix bogus test code that was breaking unrelated tests.

Author: itamar
Review: glyph
Fixes: [#5558](https://github.com/twisted/twisted/issues/5558)

Strangely, monkeypatching code in Twisted tends to break things. Stop doing that.
